runhouse/resources/resource.py

```diff
@@ -321,36 +322,47 @@ def is_local(self):
     # TODO [DG] Implement proper sharing of subresources (with an overload of some kind)
     def share(
         self,
-        users: Union[str, List[str]],
+        users: Union[str, List[str]] = None,
         access_level: Union[ResourceAccess, str] = ResourceAccess.READ,
         global_visibility: Optional[ResourceVisibility] = None,
         notify_users: bool = True,
         headers: Optional[Dict] = None,
     ) -> Tuple[Dict[str, ResourceAccess], Dict[str, ResourceAccess]]:
-        """Grant access to the resource for the list of users (or a single user). If a user has a Runhouse account they
+        """Grant access to the resource for a list of users (or a single user). If a user has a Runhouse account they
         will receive an email notifying them of their new access. If the user does not have a Runhouse account they will
-        also receive instructions on creating one, after which they will be able to have access to the Resource.
+        also receive instructions on creating one, after which they will be able to have access to the Resource. If
+        ``global_visibility`` is set to ``searchable``, users will not be notified.
```

Comment:
I agree with this statement, is it being set somewhere in this PR or am I missing it?

Reply:
yea we set it in the `share` method if specified

Reply:
are we setting the default `notify_users=True` to False when the visibility is set to public? I can't seem to find where this happens

Reply:
we don't need to since den knows how to handle that use case, but do you think we should on the OSS side anyway for clarity?

Reply:
oh ok, I also didn't see it on the den side but I could've just missed it. but if that's the case, I think just adding a comment in the oss code mentioning should be enough